content/en/docs/tutorials/configuration/configure-java-microservice/configure-java-microservice.md

```diff
@@ -6,63 +6,223 @@ weight: 10
 
 <!-- overview -->
 
-In this tutorial you will learn how and why to externalize your microservice’s configuration.
-Specifically, you will learn how to use Kubernetes ConfigMaps and Secrets to set environment
-variables and then consume them using MicroProfile Config.
-
+In this tutorial, you will learn how and why to externalize your microservice’s 
+configuration. Specifically, you will learn how to use Kubernetes ConfigMaps and 
+Secrets to set environment variables and then consume them using MicroProfile Config.
 
 ## {{% heading "prerequisites" %}}
 
-### Creating Kubernetes ConfigMaps & Secrets
+Before starting, make sure you have the following prerequisites:
+
+1. **Basic understanding of Kubernetes concepts**
+
+   If you're new to Kubernetes, it's essential to familiarize yourself with its core 
+   concepts. You can get started by exploring the [Kubernetes Basics](/docs/tutorials/kubernetes-basics/) 
+   tutorial.
+
+2. **A running Kubernetes cluster**
+
+   Set up a Kubernetes cluster, you can use [Minikube documentation](https://minikube.sigs.k8s.io/docs/start/) 
+   to learn more about running a Kubernetes cluster.
+
+3. **Java development environment set up on your local machine**
+
+   Ensure you have Java installed. You can check by running `java -version` in your 
+   terminal. If not installed, you can refer to the official website of the 
+   [OpenJDK](https://openjdk.java.net/).
+
+Now that you have the prerequisites in place, let's proceed with the tutorial.
+
+
+## Creating Kubernetes ConfigMaps & Secrets
+
+There are several ways to set environment variables for a container in Kubernetes. 
+In this tutorial section, you'll explore how to use ConfigMaps and Secrets to manage
+your microservice's environment variables.
+
+### ConfigMaps: Storing Non-Sensitive Data
+
+ConfigMaps are perfect for storing non-sensitive configuration data that can be 
+shared across multiple containers. They are Kubernetes API Objects that store 
+non-confidential key-value pairs.
+
+#### Step 1: Create a ConfigMap
+
+To get started, let's create a ConfigMap to store your microservice's application name. 
+Follow these steps:
+
+1. Open a text editor of your choice.
+
+2. Create a new file with a `.yaml` extension, for example, `myapp-configmap.yaml`.
+
+3. Write the following content into the `myapp-configmap.yaml` file:
+
+    ```yaml
+        apiVersion: v1
+        kind: ConfigMap
+        metadata:
+          name: myapp-config
```

Comment:
Please adjust the indentation accordingly. For numbered list, always indent the text and the enclosed paragraphs by 3 spaces. Avoiding 4-space indentation is important because indentation by 4 spaces is treated automatically by the markdown parser as code snippet. We want explicit markups for code snippets rather than this kind of implicit styling.


```suggestion
   ```yaml
   apiVersion: v1
   kind: ConfigMap
   metadata:
     name: myapp-config
```

Reply:
Hi @tengqm firstly sorry for the late reply on this PR due to exam preparation and work. I have made changes suggested by you and @sftim. Please do let me know if something else is required.
I also tried replacing code snippet with code shortcode as suggested by @sftim but there was netlify check issues with msg "non-zero exit code-2". So, I reverted back the code shortcode.